ouroboros-consensus-test/src/Test/Util/ChainDB.hs

```diff
@@ -0,0 +1,179 @@
+  , cdbCheckInFuture = CheckInFuture $ \vf -> pure (VF.validatedFragment vf, [])
+    -- ChainDB <--- TODO
+    -- StateMachine: InFuture.miracle (readTVar varCurSlot) maxClockSkew
+    -- ThreadNet: InFuture.reference (configLedger cfg)
+    --                               InFuture.defaultClockSkew
+    --                               (OracularClock.finiteSystemTime clock)
+    -- BlockFetch: CheckInFuture $ \vf -> pure (validatedFragment vf, [])
+    -- FollowerPromptness: CheckInFuture $ \vf -> pure (validatedFragment vf, [])
+
+    -- It seems that this allows for splitting the past from future: the miracle function assumes an
+    -- oracle that can always give us the 'correct' slot. The FollowerPromptness tests and the
+    -- BlockFetch tests assume that everything is in the past, so no behaviour based on the future.
+
+  , cdbImmutableDbCacheConfig = ImmutableDB.CacheConfig 2 60 -- ImmutableDB
+
+  -- Cache configuration for the immutable DB. How many past chunks to cache (keep in memory,
+  -- excluding the current), and when to expire past chunks.
+
+  -- StateMachine: Index.CacheConfig 2 60
+  -- ThreadNet: Index.CacheConfig 2 60
+  -- BlockFetch.Client: ImmutableDB.CacheConfig 2 60
+  -- FollowerPromptness: ImmutableDB.CacheConfig 2 60
+
+  -- Misc
+  , cdbTracer = tracer -- ChainDB, ImmutableDB, VolatileDB, LgrDB
+  , cdbTraceLedger = traceLedger -- LgrDB
+  -- StateMachine
+  --   cdbTracer                 = tracer
+  --   cdbTraceLedger            = nullTracer
+  -- ThreadNet more complicated, since we use tracers for testing
+  -- BlockFetch
+  --   cdbTraceLedger            = nullTracer
+  --   where tracer is passed to the constructor
+  --   cdbTracer = Tracer \case
+  --      ChainDBImpl.TraceAddBlockEvent ev ->
+  --         traceWith tracer $ "ChainDB: " <> show ev
+  --      _ -> pure ()
+  -- FollowerPromptness uses BlockFetch tracers
+
+  , cdbRegistry = registry -- ChainDB, ImmutableDB
+
+  , cdbGcDelay = 1 -- ChainDB, size of the bounded queue
+  , cdbGcInterval = 1 -- ChainDB
+  , cdbBlocksToAddSize = 1 -- ChainDB
+  -- FollowerPromptness:
+  --    cdbBlocksToAddSize        = 1
+  --    cdbGcDelay                = 1
+  --    cdbGcInterval             = 1
+  -- BlockFetch
+  --    cdbBlocksToAddSize        = 1
+        -- not relevant for this test
+  --    cdbGcDelay                = 1
+  --    cdbGcInterval             = 1
+  -- StateMachine
+  --  , cdbBlocksToAddSize        = 2
+      -- We don't run the background threads, so these are not used
+  --  , cdbGcDelay                = 1
+  --  , cdbGcInterval             = 1
+  }
+
+
+-- background threads:
+--   addBlockThread -- moves blocks synchronously from the queue to the "ChainDB"
+--   gcThread -- volatile to immutable (?)
+--   copyAndSnapshotThread --
+--
+--
+    -- epochSize = eraEpochSize $ topLevelConfigLedger cfg
+    -- Only when LedgerConfig blk ~ EraParams, which is not likely to hold for all cases.
```

Comment:
If you want to slightly widen the scope of this PR: It might also be useful to add functionality related to running the ChainDB, i.e. multiple tests use `openDBInternal` in order to just run the `intAddBlockThread` background thread.

Reply:
We decided not to go ahead with this in this PR -- it might be premature to factor out this functionality.